app/admin/user.rb

```diff
@@ -53,4 +53,22 @@
       row :updated_at
     end
   end
+
+  action_item :impersonate_user, only: :show do
+    params = {
+      user_id: resource.id.to_s,
+      admin_user_id: current_admin_user.id.to_s,
+      expiry: 1.hour.from_now,
```

Comment:
I'd go with something more like 5 minutes I guess

Reply:
I think the expiration can be set at the message level instead of having this custom attribute 

>Making messages expire
By default messages last forever and verifying one year from now will still return the original value. But messages can be set to expire at a given time with :expires_in or :expires_at.
>
>crypt.encrypt_and_sign(parcel, expires_in: 1.month)
crypt.encrypt_and_sign(doowad, expires_at: Time.now.end_of_year)
Then the messages can be verified and returned up to the expire time. Thereafter, verifying returns nil.

https://api.rubyonrails.org/classes/ActiveSupport/MessageEncryptor.html

Reply:
oh I see you are using `expiry` for setting it in the actual user token, so then I don't see why we would need to pass the value around, you can do it in the users controller right away